### PR TITLE
[CONDA] Revamp conda recipe.

### PR DESCRIPTION
**Pull request by tqchen**
- Combines two packages into a single recipe.
- Enable windows build.
- Better packaging hash tag (use git string).

### CODE REVIEW

**Comment by tqchen** [comment]
cc @junrushao1994 @abergeron

**Comment by junrushao1994** [comment]
Nice! I am a big fan of conda! Will take a look tonight!

File: conda/recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    string: {{ build_tag }}
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake
+        - git
+        - make # [not win]
+      host:
+        - zlib
+        - llvmdev ==10.0.0
```
Comment (junrushao1994) [inline conda/recipe/meta.yaml:62]: why not
```suggestion
        - llvmdev == 10.0.0
```

File: docker/Dockerfile.conda_cuda100 [hunk 2]
```diff
@@ -17,30 +17,16 @@
...
+RUN bash /install/ubuntu_install_conda.sh
 
 ENV PATH /opt/conda/bin:$PATH
 ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
-ENV CONDA_BLD_PATH /tmp
-
-WORKDIR /workspace
-RUN chmod -R a+w /workspace
\ No newline at end of file
+ENV CONDA_BLD_PATH /tmp
```
Comment (junrushao1994) [inline docker/Dockerfile.conda_cuda100:32]: add EOL

File: docker/Dockerfile.conda_cuda90 [hunk 3]
```diff
@@ -17,30 +17,16 @@
...
+RUN bash /install/ubuntu_install_conda.sh
 
 ENV PATH /opt/conda/bin:$PATH
 ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
-ENV CONDA_BLD_PATH /tmp
-
-WORKDIR /workspace
-RUN chmod -R a+w /workspace
\ No newline at end of file
+ENV CONDA_BLD_PATH /tmp
```
Comment (junrushao1994) [inline docker/Dockerfile.conda_cuda90:32]: add EOL

**Review by junrushao1994 — COMMENTED** [review]
mainly some formatting nitpicks

File: conda/recipe/meta.yaml (hunk 1 above)
Comment (tqchen) [reply conda/recipe/meta.yaml:62]: this is conda specific syntax(space will not work)

Comment (junrushao1994) [reply conda/recipe/meta.yaml:62]: interesting...got it!

**Comment by tqchen** [comment]
Thanks @junrushao1994 updated

**Review by junrushao1994 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**